processor/src/fast/basic_block.rs

```diff
@@ -119,7 +119,7 @@ impl FastProcessor {
             self.execute_decorator(decorator, host)?;
         }
 
-        self.execute_after_exit_decorators(node_id, program, host)
+        self.execute_after_exit_decorators(node_id, current_forest, host)
```

Comment:
You're absolutely right - we should only execute decorators in debug mode. This is exactly what issue #1821 is about. The goal there is to remove the debug mode flag from the assembler and always assemble with debug info, but the processor should only execute decorators when in debug mode.

Your suggested API looks great:
```rust
if self.in_debug_mode {
    for decorator in current_forest.get_node_decorators(node_id) {
        self.execute_decorator(decorator, host)?;
    }
}
```

This work is tracked in #1821 "always include debug info when assembling" and we plan to implement the debug mode conditional execution as part of that issue. We'll make sure the fast processor only pays the decorator execution cost when debugging.